src/WireReader.cs

```diff
@@ -201,11 +201,11 @@ public string ReadDomainName()
 
             // Read current label and remaining labels.
             var buffer = ReadBytes(length);
-            if (buffer.Any(c => c > 0x7F))
-            {
-                throw new InvalidDataException("Only ASCII characters are allowed.");
-            }
-            var name = Encoding.ASCII.GetString(buffer, 0, length);
+            //if (buffer.Any(c => c > 0x7F))
```

Comment:
Why are you removing this check. The spec only allows US-ASCII.

Reply:
yes, when you Encoding.ASCII.GetString - it replaces all non ASCII with "?", so looks legal to just send no answer. But I left it for writing because then it illegal :)

Reply:
Any way https://tools.ietf.org/html/rfc7997

Reply:
RFC 7997 does not apply to DNS.  DNS strings and labels have very precise requirements.

Please restore the checks.